examples/list-syntax.kl

```diff
@@ -3,6 +3,22 @@
 (import "defun.kl")
 
 
+(defun pair-list-syntax (head tail stx)
+  (cons-list-syntax head
+    (cons-list-syntax tail
+      '()
+      stx)
+    stx))
+
+(defun append-list-syntax (list tail stx)
+  (syntax-case list
+    [()
+     tail]
+    [(cons car cdr)
+     (cons-list-syntax car
+       (append-list-syntax cdr tail stx)
+       stx)]))
```

Comment:
I added these two new helper functions in order to make it easier to write the code-generation code. I'm not using anything else from this module, but this seemed like the best place to put them.